diff -up kalzium-23.08.0/src/solver/modwrap.c.1~ kalzium-23.08.0/src/solver/modwrap.c
--- kalzium-23.08.0/src/solver/modwrap.c.1~	2023-08-23 18:23:14.632780602 +0200
+++ kalzium-23.08.0/src/solver/modwrap.c	2023-08-23 18:25:48.111060709 +0200
@@ -13,7 +13,7 @@
 
 char *solve_equation(const char *eq)
 {
-    static value *solve_equation_closure = NULL;
+    static const value *solve_equation_closure = NULL;
     if (solve_equation_closure == NULL) {
         solve_equation_closure = caml_named_value("solve_equation");
     }
